src/codeflare_sdk/utils/generate_yaml.py

```diff
@@ -44,31 +46,160 @@ def gen_names(name):
         return name, name
 
 
-def update_dashboard_route(route_item, cluster_name, namespace):
-    metadata = route_item.get("generictemplate", {}).get("metadata")
-    metadata["name"] = f"ray-dashboard-{cluster_name}"
-    metadata["namespace"] = namespace
-    metadata["labels"]["odh-ray-cluster-service"] = f"{cluster_name}-head-svc"
-    spec = route_item.get("generictemplate", {}).get("spec")
-    spec["to"]["name"] = f"{cluster_name}-head-svc"
+# Check if the ingress api cluster resource exists
+def is_openshift_cluster():
+    try:
+        config_check()
+        api_instance = client.CustomObjectsApi(api_config_handler())
+        api_instance.get_cluster_custom_object(
+            "config.openshift.io", "v1", "ingresses", "cluster"
+        )
 
+        return True
+    except client.ApiException as e:
+        if e.status == 404:
+            return False
+        else:
+            print(f"Error detecting cluster type defaulting to Kubernetes: {e}")
+            return False
 
-# ToDo: refactor the update_x_route() functions
-def update_rayclient_route(route_item, cluster_name, namespace):
-    metadata = route_item.get("generictemplate", {}).get("metadata")
-    metadata["name"] = f"rayclient-{cluster_name}"
-    metadata["namespace"] = namespace
-    metadata["labels"]["odh-ray-cluster-service"] = f"{cluster_name}-head-svc"
-    spec = route_item.get("generictemplate", {}).get("spec")
-    spec["to"]["name"] = f"{cluster_name}-head-svc"
+
+def generate_default_ingresses(
+    cluster_name, namespace, ingress_domain, local_interactive
+):  # pragma: no cover
+    dir = pathlib.Path(__file__).parent.parent.resolve()
+    with open(f"{dir}/templates/ingress-template.yaml.tmpl", "r") as template_file:
+        ingress_template = Template(template_file.read())
+
+    if is_openshift_cluster():
+        try:
+            config_check()
+            api_client = client.CustomObjectsApi(api_config_handler())
+            ingress = api_client.get_cluster_custom_object(
+                "config.openshift.io", "v1", "ingresses", "cluster"
+            )
```

Comment:
Correct me if I'm wrong on the logic here. These checks seem they are already performed as part of the `is_openshift_cluster()` func